src/Type/Definition/BooleanType.php

```diff
@@ -32,22 +34,30 @@ public function serialize($value)
     /**
      * @param mixed $value
      * @return bool
+     * @throws Error
      */
     public function parseValue($value)
     {
-        return is_bool($value) ? $value : Utils::undefined();
+        if (is_bool($value)) {
+            return $value;
+        }
+
+        throw new Error("Cannot represent value as boolean: " . Utils::printSafe($value));
     }
 
     /**
-     * @param $valueNode
+     * @param Node $valueNode
      * @param array|null $variables
      * @return bool|null
+     * @throws \Exception
      */
     public function parseLiteral($valueNode, array $variables = null)
     {
         if ($valueNode instanceof BooleanValueNode) {
             return (bool) $valueNode->value;
         }
-        return Utils::undefined();
+
+        // Intentionally without message, as all information already in wrapped Exception
+        throw new \Exception();
```

Comment:
It throws an empty exception as otherwise the error messages do not match anymore with the expected error messages copied from graphql-js. Not super nice, but the other solution would have been updating all tests and diverging from graphql-js in that sense.